docs/errors/E0717.md

```diff
@@ -0,0 +1,46 @@
+# E0717: using a '.' after a '?.' might fail, since '?.' might return 'undefined'
+
+```config-for-examples
+{
+  "globals": {
+  }
+}
+```
+
+In Javascript, the `?.` operator performs a conditional access only if the left-hand side is not `null` or `undefined`. If the left-hand side is `null` or `undefined`, the expression returns `undefined` and the right-hand side is not evaluated. Using it suppresses the error that would otherwise be thrown if the left-hand side were `null` or `undefined` and the operator `.` were used instead. However, if the left-hand side is not `null` or `undefined`, the expression returns the value of the right-hand side, which may be `null` or `undefined`. If the right-hand side is `null` or `undefined`, using the `.` operator will throw an error:
```

Comment:
Nit: I think this explanation is wordy. A beginner is probably going to be confused by so many words.

I think this explanation focuses on what's important for the diagnostic:

> In JavaScript, `x?.y` ignores `y` and returns `undefined` if `x` is `null` or `undefined`. If `?.` returns `undefined`, then using `.` after will throw an error: